### PR TITLE
fix: fade list dividers on hover across the remaining settings lists

### PR DESCRIPTION
**Pull request by cyberk1ng**
## Summary by CodeRabbit

* **New Features**
  * Updated row divider behavior in Categories, Labels, Habits, Measurables, Dashboards, and Matrix maintenance: divider hairlines now fade on hover while the row layout stays fixed.
* **Bug Fixes**
  * Prevented divider lines from visually bisecting the currently hovered row.
* **Documentation**
  * Clarified the hover-divider behavior in settings documentation and release notes.
* **Tests**
  * Added/expanded widget tests to verify hover fade behavior (including edge cases like first/last rows and pointer leave).

### CODE REVIEW

File: test/test_utils/hover_divider_harness.dart [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+/// `HoverDividerIndex` has to stay robust against. The pointer is removed
+/// automatically at the end of the test.
+Future<TestGesture> hoverListRow(WidgetTester tester, Finder row) async {
+  final gesture = await tester.createGesture(kind: PointerDeviceKind.mouse);
+  await gesture.addPointer(location: Offset.zero);
+  addTearDown(gesture.removePointer);
+  await gesture.moveTo(tester.getCenter(row));
+  await tester.pump();
+  return gesture;
+}
```
[the remark this answers is not in the document]
Comment (cyberk1ng) [reply test/test_utils/hover_divider_harness.dart:64]: Fixed, though not quite as suggested — the pointer now parks off-screen at `(-1, -1)` for both the initial `addPointer` and `unhoverRows`, rather than starting at the row centre.

Your underlying point was right: `Offset.zero` is the top-left *inside* the view, so the add briefly hovered whatever chrome sits there (back button, header title).

I kept the `moveTo` rather than adding the pointer directly at the row. I tested your exact suggestion first and it does pass — `addPointer(location:)` reports hover on its own, so this isn't a correctness issue either way. But entering by motion is what a real pointer does, and it keeps `hoverListRow` identical to the retarget move a test makes when it continues the same gesture onto the next row (`moving between rows retargets the fade`), so the enter path under test is the same one in both cases. Parking off-screen addresses the flakiness concern without giving that up.

File: test/test_utils/hover_divider_harness.dart [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+/// Moves [gesture] away from every row, clearing the hover state.
+Future<void> unhoverRows(WidgetTester tester, TestGesture gesture) async {
+  await gesture.moveTo(Offset.zero);
+  await tester.pump();
+}
```
[the remark this answers is not in the document]
Comment (cyberk1ng) [reply test/test_utils/hover_divider_harness.dart:70]: Fixed — `unhoverRows` now moves to `(-1, -1)` instead of `Offset.zero`, exactly as suggested.

Shared the offset with `hoverListRow`'s initial `addPointer` as a named `_pointerParked` constant, since both had the same problem: the origin is inside the view where a back button or header title sits, so parking there hovers real chrome and lets it leak into tests about list rows.

All 182 tests across the affected suites still pass.
